### DISCUSSION TITLE
Can't get bg overlay to work in scss

### DISCUSSION THREAD

**Discussion by dn3hgh** [community/community#107479] [Programming Help]
### Body

First time working with .scss. Everything seems to be good including the background image. but as soon as I try the .overlay inside main.scss I can't get a solid color to overlay the image to then work with opacity. Thanks!

**here is my repository for this**...(https://github.com/dn3hgh/Personal-Portfolio.git)

**main.scss**
`$primary-color: #101820;
$secondary-color: #FFB612;
$tertiary-color: #C60C30;
$link-color: #003087;
$neutral-color: #a5acaf;
$show-home-image: true;
$home-image: url(../img/background.jpg);
$background-opacity: 0.9;

@mixin easeOut {
    transition: all 0.5s ease-out;
}

@mixin background {
    @if $show-home-image {
        &#bg-img {
            background: $home-image;
            background-attachment: fixed;
            background-size: cover;
            

        .overlay {
            position: absolute;
            top: 0;
            right: 0;
            width: 100%;
            height: 100%;
            background: $primary-color;
            
        }
        }
    }
}

*   {
    box-sizing: border-box;
}

body{
    @include background;
    background-color: $primary-color;
    color: #fff;
    height: 100%;
    margin: 0;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
    line-height: 1.5;
}

//headings
h1, h2, h3 {
    margin: 0;
    font-weight: 400;

    &.lg-heading    {
        font-size: 6rem;
    }

    &.sm-heading    {
        margin-bottom: 2rem;
        padding: 0.2rem 1rem;
        background: rgba(lighten($color: $primary-color, $amount: 2), 0.5);
    }

}

a {
    color: #ffffff;
    text-decoration: none;
}

header {
    position: fixed;
    z-index: 2;
    width: 100%;

}

.text-secondary {
    color: $secondary-color;
}

main    {
    padding: 4rem;
    height: 100%;

    .icons{
        margin-top: 1rem;

        a {
            padding: 0.4rem;
        
            &:hover {
                color: $tertiary-color;
                @include easeOut();
            }
        }
    }

    &#home {
        overflow: hidden;
        h1 {
            margin-top: 20vh;
        }
    }
}

`

**HTML**
`<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link
      rel="stylesheet"
      href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css"
      integrity="sha512-DTOQO9RWCH3ppGqcWaEA1BIZOC6xxalwEsw9c2QQeAIftl+Vegovlnee1c9QX4TctnWMn13TZye+giMm8e2LwA=="
      crossorigin="anonymous"
      referrerpolicy="no-referrer"
    />

    <link rel="stylesheet" href="/css/main.css" />

    <title>Welcome to my portfolio</title>
  </head>
  <body id="bg-img">
    
    <header>
      <div class="menu-btn">
        <div class="btn-line"></div>
        <div class="btn-line"></div>
        <div class="btn-line"></div>
      </div>

      
      <nav class="menu">
        <div class="menu-branding">
          <div class="portrait"></div>
        </div>
        <ul class="menu-nav">
          <li class="nav-item">
            <a href="/" class="nav-link">HOME</a>
          </li>
          <li class="nav-item">
            <a href="/about.html" class="nav-link">ABOUT ME</a>
          </li>
          <li class="nav-item">
            <a href="/work.html" class="nav-link">MY WORK</a>
          </li>
          <li class="nav-item">
            <a href="/contact.html" class="nav-link">HOW TO REACH ME</a>
          </li>
        </ul>
      </nav>
    </header>

    
    <main id="home">
      <h1 class="lg-heading">
        Daniel <span class="text-secondary">Hughes</span>
      </h1>
      <h2 class="sm-heading">
        Web Developer and Wordpress for small businesses.
      </h2>

      <div class="icons">
        <a href="#!">
          <i class="fab fa-twitter fa-2x"></i>
        </a>
        <a href="#!">
          <i class="fab fa-linkedin fa-2x"></i>
        </a>
        <a href="#!">
          <i class="fab fa-github fa-2x"></i>
        </a>
      </div>
    </main>

    <script src="js/main.js"></script>
  </body>
</html>
`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ghasemizade** [comment]
Hi @dn3hgh 👋

In this part of the code, I've replaced the usage of opacity with rgba to set the background color. Additionally, I've used z-index to control how the background is displayed in different layers:

## scss
```
.overlay {
  position: absolute;
  top: 0;
  right: 0;
  width: 100%;
  height: 100%;
  background-color: rgba($primary-color, $background-opacity);
  z-index: -1; /* To place the background behind other elements */
}

```

Additionally, open a `<div>` tag alongside the `<body>` tag and assign the class `overlay` to it:

## html
```
<body>
  <div class="overlay"></div> <!-- This div will have the overlay class -->
  <!-- Your existing HTML content here -->
</body>

```

This `<div>` will serve as the overlay element, covering the entire page with the specified background color and opacity.

Thank you.
